src/vcf/record.cpp

```diff
@@ -274,9 +289,54 @@ namespace ebi
         }
     }
 
+    void Record::check_predefined_tag(std::string const & field_key, std::string const & field_value, std::vector<std::string> const & values,
+                                      std::map<std::string, std::pair<std::string, std::string>> const & tags) const
+    {
+        auto iterator = tags.find(field_key);
+        if (iterator != tags.end()) {
+            try {
+                check_field_cardinality(field_key, values, iterator->second.second);
+                check_field_type(values, iterator->second.first);
+            } catch (std::shared_ptr<Error> ex) {
+                raise(std::make_shared<Error>(line, field_key + "=" + field_value
+                                + " is not compliant with the VCF specification" + ex->message));
```

Comment:
The error message thrown would be something like `INFO AC=2,3,4,5 is not complaint with the VCF specification Number=G. Contains 4 values, expected 2` 
I'm not sure about the message, especially this `Number=X` or `Type=Y` part, but I added it to avoid some problem in the code and also because adding this would prompt the user to check that the predefined tag does not contain a meta definition and it would be good to add it. Or say, they will get to know that the Number has been defined as A, so that's why the expected number of values is 2 (the number of alt alleles present in the line).